code_examples/sdk_examples/src/getFunds.ts

```diff
@@ -1,68 +1,27 @@
 import * as Kilt from '@kiltprotocol/sdk-js'
+import { BalanceUtils, Blockchain } from '@kiltprotocol/chain-helpers'
+import type {
+  KiltAddress,
+  MultibaseKeyPair,
+  TransactionSigner
+} from '@kiltprotocol/types'
+import { Multikey } from '@kiltprotocol/utils'
 import { BN } from '@polkadot/util'
-import { setTimeout } from 'timers/promises'
-
-async function failproofSubmit(
-  tx: Kilt.SubmittableExtrinsic,
-  submitter: Kilt.KeyringPair
-) {
-  try {
-    await Kilt.Blockchain.signAndSubmitTx(tx, submitter)
-  } catch {
-    // Try a second time after a small delay and fetching the right nonce.
-    const waitingTime = 12_000 // 12 seconds
-    console.log(
-      `First submission failed for workshop. Waiting ${waitingTime} ms before retrying.`
-    )
-    await setTimeout(waitingTime)
-    console.log('Retrying...')
-    // nonce: -1 tells the client to fetch the latest nonce by also checking the tx pool.
-    const resignedBatchTx = await tx.signAsync(submitter, { nonce: -1 })
-    await Kilt.Blockchain.submitSignedTx(resignedBatchTx)
-  }
-}
 
+// TODO: Look into tidier way with PD Keyring…
```

Comment:
an idea could be to make a change to the sdk so that we also return a polkadot address from generateKeypair ?

Reply:
I am not sure this is a question i can answer @rflechtner and I can't actually quite remember what this was in reference to. I leave it up to you :)